plugins/identity-shiro/src/main/java/org/opensearch/identity/shiro/ShiroTokenManager.java

```diff
@@ -51,15 +54,16 @@ public Optional<AuthenticationToken> translateAuthToken(org.opensearch.identity.
             final BasicAuthToken basicAuthToken = (BasicAuthToken) authenticationToken;
             return Optional.of(new UsernamePasswordToken(basicAuthToken.getUser(), basicAuthToken.getPassword()));
         }
-
         return Optional.empty();
     }
 
     @Override
-    public AuthToken issueToken(String audience) {
+    public AuthToken issueOnBehalfOfToken(Subject subject, OnBehalfOfClaims claims) {
```

Comment:
Just as suggestion, do you think we could generalize this method a bit (introducing `Claims` interface):

```
```suggestion
    public AuthToken issueToken(Subject subject, Claims claims) {
        ....
    }
```

I suspect the on behalf of won't be the only tokens we would manage

Reply:
I had a more generic claims interface but @peternied suggested we wait for the actual use before making things too generic. Does that sound alright?

Reply:
> @peternied suggested we wait for the actual use before

Sure, you guys drive that, no objections